docs/src/contributing/git-workflow.md

```diff
@@ -51,36 +101,147 @@ If the branch exists in a triton/neptune official repo, (as opposed to a persona
 
 So if working on issue `#232` and adding feature *walk_and_chew_gum* one might name the branch `myuser/232_walk_and_chew_gum_pr`.
 
-### release branch
+### Testnet policy (pre-release)
+
+It is project policy that each major and minor release be tested as a
+release-candidate in the public community testnet prior to making an official
+release.
+
+A period of public testnet testing provides the community with an opportunity to
+try out candidate builds in their own unique environments and report any issues,
+and to prepare for new features in advance.
+
+It is also an important period of integration testing "in the wild" for the
+release candidate binary.  Operating on a public testnet, it will necessarily be
+exposed to peers running older versions of the software and may shake out issues
+that do not occur with automated testing.
+
+#### Some specifics
+
+1. The release candidate should be published using the same release process as
+   the eventual release; see [releasing](releasing.md).
+2. Each release candidate must be publicly announced with a dedicated topic in
+   [talk.neptune.cash](https://talk.neptune.cash).
+3. The release candidate should target 2 weeks of testing after the announcement
+   on average with a minimum of 1 week.
+4. The company behind neptune-cash will provide at least 1 dedicated machine for
+   the public testnet for the purpose of running the release-candidate binary
+   and composing blocks.  Of course the community is also encouraged to run
+   nodes and mine (compose or guess) if possible.
+5. A set of automated tests will be created that utilize the RPC API to perform
+   automated transactions and test basic functionality of the live running
+   nodes. These tests should/must be run at least once against each
+   release-candidate binary while connected to the public testnet network.
```

Comment:
A different set of tests for every release candidate, or are all release candidates subject to the same set of tests (that can be updated over time to reflect API evolution)?

Let's add this rule to the test policy after the API test suite exists. If we add it before, we risk over-committing and risk having to delay urgent updates, even if they are orthogonal to the things being tested by that test suite.

Reply:
> A different set of tests for every release candidate, or are all release candidates subject to the same set of tests (that can be updated over time to reflect API evolution)?

the latter.  It's basically an integration test suite, but running against node(s) on a live (test) network. This is the sort of thing that perhaps we could get a google-summer-of-code volunteer/intern to build, with proper guidance.

> Let's add this rule to the test policy after the API test suite exists. If we add it before, we risk over-committing and risk having to delay urgent updates, even if they are orthogonal to the things being tested by that test suite.

I'd suggest keeping it, but mark it as a goal/aspirational rather than policy, rule or promise.